internal/db/code.go

```diff
@@ -0,0 +1,114 @@
+package db
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/starlight-go/starlight"
+	"reflect"
+)
+
+type Code struct {
+	ID       uuid.UUID
+	Name     string
+	Function Function
+	Runtime  Runtime
+	Code     string
+}
+
+type Runtime int64
+
+const (
+	Golang Runtime = iota
+	Javascript
+	Starlark
+)
+
+type Function int64
+
+const (
+	Validator Function = iota
+)
+
+var nativeFunctionMap map[Code]func(interface{}) (interface{}, error) = map[Code]func(interface{}) (interface{}, error){
+	boolValidator:         boolValidatorFunc,
+	intValidator:          intValidatorFunc,
+	enumValidator:         enumValidatorFunc,
+	stringValidator:       stringValidatorFunc,
+	textValidator:         textValidatorFunc,
+	emailAddressValidator: emailAddressValidatorFunc,
+	uuidValidator:         uuidValidatorFunc,
+	floatValidator:        floatValidatorFunc,
+	URLValidator:          URLValidatorFunc,
+}
+
+func CallFunc(c Code, sf StorageFormat, arg interface{}) (interface{}, error) {
```

Comment:
This function needs a different name if that's gonna be its type signature :)

Right now this is "CallValidator" at best!